nrf_modem/doc/dectphy.rst

```diff
@@ -0,0 +1,296 @@
+Due to the nature of a radio scheduler, which allows radio operations to be executed at a specific time in the future, all radio operations in the DECT PHY interface are asynchronous and their completion is signalled to the application using callbacks.
+
+All scheduling is final, meaning that it is not possible to unschedule a scheduled operation.
+
+A radio operation may be scheduled to execute a specific time in the future, or immediately, if the radio is not currently executing any other operation.
+
+All radio operations have an application-defined handle which can be used to identify the operation and that is returned by the callback signalling the completion of an operation.
+The operation handle is entirely opaque to all underlying layers, including the DECT PHY interface.
+
+Modem time and operation latency
+--------------------------------
+
+Operation execution is scheduled by the application according to the modem time, which is a 64 bit counter kept by the modem.
+All radio operation callbacks provide the value that the modem time counter had at the moment the callback was sent by the modem core to the application core, thus providing a way for the application to track the modem time without explicitly querying the modem for it.
+
+If necessary, the application can retrieve the modem time counter value by calling the :c:func:`nrf_modem_dect_phy_time_get` function.
+
+Both radio state transitions between operations in the modem core and the inter-processor communication (IPC) mechanism between the application and more cores have latency.
+The latency of both radio state transitions and IPC is measured and provided by the DECT PHY interface upon initialization, by the :c:member:`nrf_modem_dect_phy_callbacks.init` callback function, in the :c:member:`nrf_modem_dect_phy_modem_cfg.latency` parameter.
+
+.. note::
+   The application must account for operations' latency when scheduling operations.
+
+Temperature monitoring
+======================
+
+The DECT PHY interface reports the current SoC temperature as measured by the DECT NR+ modem firmware in all radio operation callbacks, and upon initialization.
+This allows the application to track the changes in temperature and adjust further scheduling of operations accordingly, to ensure the device remains within safe operating temperatures.
+
+The DECT NR+ modem firmware has an internal temperature protection mechanism that prevents the SoC from operating above safe temperature limits.
+The operating temperature limit is reported upon initialization by the :c:member:`nrf_modem_dect_phy_callbacks.init` callback function, in the :c:member:`nrf_modem_dect_phy_modem_cfg.temperature_limit` parameter.
+
+If the temperature threshold is reached, the modem cancels all scheduled radio operations and reports the :c:enum:`nrf_modem_dect_phy_err.NRF_MODEM_DECT_PHY_ERR_OP_CANCELED` error in their relative callbacks.
```

Comment:
When high temp is reported in DECT PHY, the modem do not implicitly cancel anything. It's simply refuses to start any new operation. Application shall in this case de-init the DECT PHY stack. As a part of deinit-procedure all the scheduled operations are canceled if any.

Reply:
ok, I will add. thanks for clarification.